### DISCUSSION TITLE
Beta: Settings

### DISCUSSION THREAD

**Discussion by roaldarbol** [mooch443/trex#224] [Questions] [answered]
Hi again @mooch443, just here with more questions, but nothing urgent. ;-)

- For the new version, do you have a list of the parameters? 
- Does it still work with a `.settings` file?
- Does it work more or less in the same way through the command line?

I might attempt to do some batch processing with it, but in the first place I'll just want to attempt a single file through the command line (then I can probably figure may way around it from there).

**Comment by mooch443** [comment, marked as answer]
You can generate a list of parameters yourself by passing `-task rst -d /output/folder` you'll get a .rst file containing the parameters and descriptions. It's a markdown-like file, but you could theoretically compile it to HTML I guess. There may be some in there that don't do anything anymore!

Yea, settings files didn't change. In fact it saves one if you convert. It also maintains internal storage for the recently opened files if you launch without parameters (`trex`).

Yes, the command-line largely remains the same. One thing that changed is that by default, if not specified otherwise, it will assume the output folder is the folder the input video is in. You can open

```
trex -i /path/to/video.mp4 -m model.pt
```

And it will save to `/path/to/video.pv`, and then drop you into tracking view after its done or you stop it. If you then run the same command again, it will directly drop you into tracking view because the .pv file already exists (if it can be read). You can force reconvert by passing `-task convert`, or try to only open by passing `-task track` which would then result in an error if the .pv does not exist yet.

These are probably the main changes.

**Reply by mooch443** [reply]
Oh btw, I updated once or twice today because there were some crashes on Windows :-)

**Reply by roaldarbol** [reply]
Amazing, I'll have a play with it! Just to understand correctly - I think you at some point mentioned that you were planning to have it not be a 2-step process (or at least not on the user-facing side), so I imagined that you just input an `.mp4` file, and you'll get tracks out - the `.pv` files wouldn't be exposed or saved. Well I guess, maybe intermediately saved... Is it automatically removed afterwards or should I do this manually? (The reason I'm asking is I've got ~700GB videos to track, and that balloons to double with `.pv` files).

What are the important parameters to specify now that the detection relies not on normal background subtraction, but on the yolo version?

I'm genuinely super super excited Tristan, I've been checking the commit updates on the interface/dev branch almost daily for months :P

**Reply by mooch443** [reply]
I mean, I never said there would be no pv files. And no, you shouldn't remove it, unless you want to overwrite it or are done with the project / exporting tracking data. It is a single step process if you just want the trajectories - it will still create a PV file, which you can then delete if you want to. Just add `-auto_quit` to the command, and trajectories will be saved. 

Of course in that case you'd need to know the right tracking parameters beforehand. If you'd change those mid-convert (which you can) it wouldn't apply to the already processed ones :) But this kind of experimentation is why pv files are kept - you can just open them up again (with the same command, with the mp4 file as input and no additional task parameter) and change some tracking stuff, save config / export again.

The most important parameters are just typical ML parameters like `detect_iou_threshold` which determines how much stuff is allowed to overlap, or `detect_conf_threshold` which tells it how sure it's supposed to be about objects. Filtering (which used to be `blob_size_ranges`) has become `track_size_filter` and `segment_size_filter` (for initial conversion step, rarely has to be set). `detect_model` of course sets the path to the .pt model file.

Haha, good to hear - I hope it won't be a disappointment then!

**Reply by mooch443** [reply]
Also, keep the comments coming - if you find some crashes, or non-functional stuff, or other ideas - let me know. Greatly appreciated!

**Reply by roaldarbol** [reply]
> I mean, I never said there would be no pv files.
I think that's just been my interpretation of things. ;-)

I'll just remove .pv files subsequently - I do that now too, so it's not a big change (I also use `-auto_quit`).

> Of course in that case you'd need to know the right tracking parameters beforehand. If you'd change those mid-convert (which you can) it wouldn't apply to the already processed ones :) But this kind of experimentation is why pv files are kept - you can just open them up again (with the same command, with the mp4 file as input and no additional task parameter) and change some tracking stuff, save config / export again.

Makes complete sense, I think I'll fidget first to find the right parameters and then convert them into a settings file.

> The most important parameters are just typical ML parameters like detect_iou_threshold which determines how much stuff is allowed to overlap, or detect_conf_threshold which tells it how sure it's supposed to be about objects. Filtering (which used to be blob_size_ranges) has become track_size_filter and segment_size_filter (for initial conversion step, rarely has to be set). detect_model of course sets the path to the .pt model file.

I think you overestimate my knowledge of ML. :P But good to know those parameters. What are typical values for those parameters? And what is a `.pt`model, and is it something I as a user need to set a path to?

> Also, keep the comments coming - if you find some crashes, or non-functional stuff, or other ideas - let me know. Greatly appreciated!

Good to know - it's sometimes hard to know whether it's being helpful or annoying!

**Comment by roaldarbol** [comment]
I'm testing with a video of ants, but the yolo model is not picking them up. Do I need to train a model specifically for my animal of interest? Or what are the differences between the models? One thing that leads me to think that is that it does find one thing during conversion, though not one of the ants, and calls it "person".

**Reply by roaldarbol** [reply]
This is on Intel Mac only CPU btw.

**Reply by roaldarbol** [reply]
And in which case, do you have a good tutorial or recommendations for how to do that? I found my way back to the data sets on roboflow you had previously linked to and found what is probably a really good dataset for working with insects (https://universe.roboflow.com/maximilian-sittinger/insect_detect_classification_v2).

**Reply by roaldarbol** [reply]
@mooch443 Still would be super interesting to hear what the approach would be. :-)
